frame/staking/src/lib.rs

```diff
@@ -554,17 +567,18 @@ impl<T: Config> StakingLedger<T> {
 		let mut remaining_slash = slash_amount;
 		let pre_slash_total = self.total;
 
-		let era_after_slash = slash_era + 1;
-		let chunk_unlock_era_after_slash = era_after_slash + T::BondingDuration::get();
+		// for a slash_era = x, any chunk that is scheduled to be unlocked at era `x + 28` (assuming
+		// 28 is the bonding duration) onwards should be slashed.
+		let slashable_chunks_start = slash_era + T::BondingDuration::get();
 
 		// Calculate the total balance of active funds and unlocking funds in the affected range.
-		let (affected_balance, slash_chunks_priority): (_, Box<dyn Iterator<Item = usize>>) = {
-			if let Some(start_index) =
-				self.unlocking.iter().position(|c| c.era >= chunk_unlock_era_after_slash)
+		let (is_proportional, affected_balance, slash_chunks_priority) = {
```

Comment:
This `is_proportional` boolean could be explained better in the docs/comments I think.